drivers/charger/sbs_charger.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2023 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sbs_sbs_charger
+
+#include <zephyr/drivers/charger/charger_sbs_charger.h>
+#include <zephyr/drivers/charger.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+LOG_MODULE_REGISTER(sbs_charger);
+
+static int sbs_cmd_reg_read(const struct device *dev, uint8_t reg_addr, uint16_t *val)
+{
+	const struct sbs_charger_config *cfg;
+	uint8_t i2c_data[2];
+	int status;
+
+	cfg = dev->config;
+	status = i2c_burst_read_dt(&cfg->i2c, reg_addr, i2c_data, ARRAY_SIZE(i2c_data));
```

Comment:
I think it'd be correct to use `sizeof(i2c_data)` here rather than `ARRAY_SIZE`. The latter gives you the number of elements, which happens to match just because happen to be 1 bytes sized.

Reply:
ack

Reply:
still relevant

Reply:
ditto https://github.com/zephyrproject-rtos/zephyr/pull/56666#discussion_r1301778391 ack